### PR TITLE
feat(staking): LW-7817 current pools managenet

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [x] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Added:
 - current portfolio management flow
 - add pool button in staking preferences drawer

Comments:
 - when managing current portfolio and clicking add poll button then current selections get overridden by pools from drawer
 - when in the preferences drawer coming from the selections flow then remove button affects underlaying selections. 

## Testing

Unfortunately besides the new buttons we need to test the complete flow as the changes affect it in various places. Let's do the full regression of the multi delegation flow, please 🙏 

## Screenshots

image
image
image
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1705/5974491727/index.html) for [0d3d6365](https://github.com/input-output-hk/lace/pull/438/commits/0d3d6365fd8686567ab118350ea010ad9b855371)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 34     | 1      | 0       | 0     | 35    | ❌     |

### CODE REVIEW

File: packages/staking/src/features/browse-pools/PortfolioBar.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const PortfolioBar = ({ onStake }: PortfolioBarParams) => {
   const { t } = useTranslation();
   const { portfolioMutators, selectedPoolsCount } = useDelegationPortfolioStore((store) => ({
     portfolioMutators: store.mutators,
-    selectedPoolsCount: selectDraftPoolsCount(store),
+    selectedPoolsCount: store.selections.length,
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/browse-pools/PortfolioBar.tsx:15]: Nitpick:

Can we still have a selector for that? I see it's being accessed more than once.

File: packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx [hunk 2]
```diff
@@ -56,14 +56,14 @@ export const StakePoolItemBrowser = ({
     title = ticker || '-';
     subTitle = <Ellipsis className={styles.id} text={id} beforeEllipsis={6} afterEllipsis={8} />;
   }
-  const draftPortfolioLength = useDelegationPortfolioStore((state) => state.draftPortfolio.length);
-  const draftPortfolioExists = draftPortfolioLength > 0;
-  const includedInDraft = useDelegationPortfolioStore((state) => state.poolIncludedInDraft(hexId));
+  const selectionsCount = useDelegationPortfolioStore((state) => state.selections.length);
+  const selectionsNotEmpty = selectionsCount > 0;
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx:60]: Nitpick:

const selectionsNotEmpty = useDelegationPortfolioStore((state) => state.selections.length > 0);

File: packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx [hunk 3]
```diff
@@ -56,14 +56,14 @@ export const StakePoolItemBrowser = ({
...
+  const poolAlreadySelected = useDelegationPortfolioStore((state) => state.queries.isPoolAlreadySelected(hexId));
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx:61]: I left another comment about naming methods `byId` etc. in this case we use `hexId` so I see it's even more relevant to include this and potentially unify usage of id/hexId.

File: packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx [hunk 4]
```diff
@@ -46,8 +46,8 @@ export const StakePoolItemBrowser = ({
   logo,
   apy,
   onClick,
-  addToDraft,
-  removeFromDraft,
+  select,
+  unselect,
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx:50]: Nitpick/question: Would `onToggle` / `onToggleSelection` do the job?

File: packages/staking/src/features/browse-pools/stake-pools-table/StakePoolsTable.tsx [hunk 5]
```diff
@@ -31,7 +31,7 @@ export const StakePoolsTable = ({ onStake, scrollableTargetId }: StakePoolsTable
   const [sort, setSort] = useState<StakePoolSortOptions>(DEFAULT_SORT_OPTIONS);
   const [stakePools, setStakePools] = useState<Wallet.StakePoolSearchResults['pageResults']>([]);
   const [skip, setSkip] = useState<number>(0);
-  const { addPoolToDraft, removePoolFromDraft } = useDelegationPortfolioStore((state) => state.mutators);
+  const { selectPool, unselectPool } = useDelegationPortfolioStore((state) => state.mutators);
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/browse-pools/stake-pools-table/StakePoolsTable.tsx:34]: Nitpick/question: Can we please follow accessing the mutators and other top-level properties without deep destructuring?

With this approach you need to go to the destructuring place to understand it's coming from the store, not inline `useState` or sth else. I like the approach from `packages/staking/src/features/drawer/StakePoolPreferences.tsx`

File: packages/staking/src/features/store/types.ts [hunk 6]
```diff
@@ -51,30 +51,41 @@ export type CurrentPortfolioStakePool = DraftPortfolioStakePool & {
   value: bigint;
 };
 
+export enum PortfolioState {
+  Free = 'Free',
+  ConfirmingNewPortfolio = 'ConfirmingNewPortfolio',
+  ManagingCurrentPortfolio = 'ManagingCurrentPortfolio',
+}
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/store/types.ts:58]: What do you think about the following?

```suggestion
export enum PortfolioProcess {
  Idle = 'Idle',
  ManagingNewPortfolio = 'ManagingNewPortfolio',
  ManagingCurrentPortfolio = 'ManagingCurrentPortfolio',
}
```

**Review by przemyslaw-wlodek — CHANGES_REQUESTED** [review]

File: packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx (hunk 4 above)
Comment (szymonmaslowski) [reply packages/staking/src/features/browse-pools/stake-pools-table/StakePoolItemBrowser/StakePoolItemBrowser.tsx:50]:
```suggestion
  onSelect,
  onUnselect,
```

File: packages/staking/src/features/store/delegationPortfolio.ts [hunk 7]
```diff
@@ -44,19 +71,16 @@ export const useDelegationPortfolioStore = create(
...
-          if (!poolEntry) return;
-          poolEntry.weight = weight;
+          store.selections = store.selections.filter((pool) => pool.id !== id);
         }),
     },
-    poolIncludedInDraft: (id) => {
-      const { draftPortfolio } = get();
-      return !!draftPortfolio?.find((pool) => pool.id === id);
+    queries: {
+      isPoolAlreadySelected: (id) => {
```
[the remark this answers is not in the document]
Comment (mrcnk) [reply packages/staking/src/features/store/delegationPortfolio.ts]: Hex ID is fine, it's more accurate in a sense that pool has: public key, id and hex id.

**Review by przemyslaw-wlodek — APPROVED** [review]

**Review by oldGreg5 — APPROVED** [review]

**Pull request merged**